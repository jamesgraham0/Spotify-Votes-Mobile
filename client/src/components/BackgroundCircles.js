import { StyleSheet, View } from 'react-native';
import Constants from '../utils/constants';

const BackgroundCircles = () => {
    return (
        <View style={styles.container}>
            <View style={[styles.circleGreen1, {backgroundColor: Constants.SPOTIFY_GREEN}]} />
            <View style={[styles.circleGreen2, {backgroundColor: Constants.SPOTIFY_GREEN}]} />
            <View style={[styles.circleGreen3, {backgroundColor: Constants.SPOTIFY_GREEN}]} />
            <View style={[styles.circleGreen4, {backgroundColor: Constants.SPOTIFY_GREEN}]} />
            <View style={[styles.circleGreen5, {backgroundColor: Constants.SPOTIFY_GREEN}]} />

            <View style={[styles.circlePurple1, {backgroundColor: Constants.SPOTIFY_PURPLE}]} />
            <View style={[styles.circlePurple2, {backgroundColor: Constants.SPOTIFY_PURPLE}]} />
            <View style={[styles.circlePurple3, {backgroundColor: Constants.SPOTIFY_PURPLE}]} />
            <View style={[styles.circlePurple4, {backgroundColor: Constants.SPOTIFY_PURPLE}]} />
            <View style={[styles.circlePurple5, {backgroundColor: Constants.SPOTIFY_PURPLE}]} />
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        width: '100%',
        height: '100%',
    },
    circleGreen1: {
        position: 'absolute',
        width: 250,
        height: 250,
        borderRadius: 125,
        opacity: 0.4,
        top: -60,
        right: -60,
      },
      circleGreen2: {
        position: 'absolute',
        width: 50,
        height: 50,
        borderRadius: 25,
        opacity: 0.4,
        top: 150,
        right: 100,
      },
      circleGreen3: {
        position: 'absolute',
        width: 90,
        height: 90,
        borderRadius: 45,
        opacity: 0.4,
        top: 110,
        right: 130,
      },
      circleGreen4: {
        position: 'absolute',
        width: 150,
        height: 150,
        borderRadius: 75,
        opacity: 0.4,
        bottom: -20,
        left: -20,
      },
      circleGreen5: {
        position: 'absolute',
        width: 100,
        height: 100,
        borderRadius: 50,
        opacity: 0.4,
        bottom: 100,
        left: -40,
      },
      circlePurple1: {
        position: 'absolute',
        backgroundColor: 'purple',
        width: 150,
        height: 150,
        borderRadius: 75,
        opacity: 0.4,
        top: -20,
        left: -10,
      },
      circlePurple2: {
        position: 'absolute',
        backgroundColor: 'purple',
        width: 80,
        height: 80,
        borderRadius: 40,
        opacity: 0.4,
        top: 100,
        left: -10,
      },
      circlePurple3: {
        position: 'absolute',
        backgroundColor: 'purple',
        width: 30,
        height: 30,
        borderRadius: 15,
        opacity: 0.4,
        top: 30,
        left: 120,
      },
      circlePurple4: {
        position: 'absolute',
        width: 150,
        height: 150,
        borderRadius: 75,
        opacity: 0.4,
        bottom: -20,
        right: -20,
      },
      circlePurple5: {
        position: 'absolute',
        width: 100,
        height: 100,
        borderRadius: 50,
        opacity: 0.4,
        bottom: 100,
        right: -40,
      },
});

export default BackgroundCircles;