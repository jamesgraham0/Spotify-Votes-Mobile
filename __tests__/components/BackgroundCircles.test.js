import React from 'react';
import renderer from 'react-test-renderer';
import BackgroundCircles from '../../client/src/components/BackgroundCircles';

describe('BackgroundCircles', () => {
    it('should render the BackgroundCircles component', () => {
        const tree = renderer.create(<BackgroundCircles />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
