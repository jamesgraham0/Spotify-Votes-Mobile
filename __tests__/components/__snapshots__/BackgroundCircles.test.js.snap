// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundCircles should render the BackgroundCircles component 1`] = `
<View
  style={
    {
      "height": "100%",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      [
        {
          "borderRadius": 125,
          "elevation": 5,
          "height": 250,
          "opacity": 0.4,
          "position": "absolute",
          "right": -60,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": -60,
          "width": 250,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 25,
          "elevation": 5,
          "height": 50,
          "opacity": 0.4,
          "position": "absolute",
          "right": 100,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 150,
          "width": 50,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 45,
          "elevation": 5,
          "height": 90,
          "opacity": 0.4,
          "position": "absolute",
          "right": 130,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 110,
          "width": 90,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 75,
          "bottom": -20,
          "elevation": 5,
          "height": 150,
          "left": -20,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 150,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 50,
          "bottom": 100,
          "elevation": 5,
          "height": 100,
          "left": -40,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 100,
        },
        {
          "backgroundColor": "#1DB954",
          "shadowColor": "#1DB954",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 75,
          "elevation": 5,
          "height": 150,
          "left": -10,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": -20,
          "width": 150,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 40,
          "elevation": 5,
          "height": 80,
          "left": -10,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 100,
          "width": 80,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 15,
          "elevation": 5,
          "height": 30,
          "left": 120,
          "opacity": 0.4,
          "position": "absolute",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "top": 30,
          "width": 30,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 75,
          "bottom": -20,
          "elevation": 5,
          "height": 150,
          "opacity": 0.4,
          "position": "absolute",
          "right": -20,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 150,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "borderRadius": 50,
          "bottom": 100,
          "elevation": 5,
          "height": 100,
          "opacity": 0.4,
          "position": "absolute",
          "right": -40,
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 6,
          "width": 100,
        },
        {
          "backgroundColor": "#B026FF",
          "shadowColor": "#B026FF",
        },
      ]
    }
  />
</View>
`;
